src/main/java/org/opensearch/knn/index/memory/NativeMemoryAllocation.java

```diff
@@ -163,11 +164,19 @@ class IndexAllocation implements NativeMemoryAllocation {
 
         @Override
         public void close() {
-            executor.execute(() -> {
+            Runnable onClose = () -> {
```

Comment:
Oh I like this - I was always wondering if we actually need to run this in an executor context. Anecdotally from you testing, how long would this method typically run?

Reply:
I did not run numbers specifically for this block. Do you want me to get those numbers?

Reply:
The other general problem with this today is that when the cache calls the `onRemoval` method for an entry - based on the Guava's documentation - it is supposed to be blocking. 
So when the cache is trying to clean up - forking the operation on this thread enables the cache to have a different usage metric compared to what is held by native memory indexes, further exacerbating the overflow problem.

Reply:
Ran some numbers - I could only get averages.

With default code (current code) - 4.26076555024 ms averaged across nodes 
With modified code - 1.63179895861 ms averaged across nodes  (Possibly because the block is sync already)